### PR TITLE
Feature - Add support for managing service tokens

### LINKED ISSUE #124: Feature - Support creation/deletion of Service Tokens
Feature request to support the creation/deletion of service tokens.

CRUD endpoints:
- Create (2 steps)
  - [Create Service Token](https://documenter.getpostman.com/view/14183654/UyxohieD#e871dc63-abcd-44e3-9f6d-46baa7aa1eac)
  - and [Assign Service Token Permissions](https://documenter.getpostman.com/view/14183654/UyxohieD#540d5ac6-6a59-4174-afb2-98ad486e6d60)
    - the format is the same as [for group permissions](https://github.com/GtheSheep/terraform-provider-dbt-cloud/pull/122), so we should be able to leverage some code
- Read
  - [Retrieve Service Token](https://documenter.getpostman.com/view/14183654/UyxohieD#e2f782ef-64a1-4582-ab4c-1bd5b2c3ffd9) with its permissions
- Update
  - We should not allow updating a Service Token. Once created it can only be Read or Deleted
- Delete
  - Not documented, but a simple `DELETE` works fine

### PR DESCRIPTION
**Pull request by b-per**
Fixes #124 

Will most likely have a small merge conflict with #139 
We will also need to know if we want to merge https://github.com/GtheSheep/terraform-provider-dbt-cloud/pull/133 before or after as this will require some slight docs rework

### datasource acceptance test passing
SCR-20230516-nmbd

### resource acceptance test passing
SCR-20230516-qwci

### CODE REVIEW

**Comment by b-per** [comment]
@GtheSheep , let me know if you need my help fixing merge conflicts once one of #139 or #140 gets merged

File: examples/resources/dbt_cloud_service_token/resource.tf [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+    }
+    service_token_permissions {
+        permission_set = "job_admin"
+        all_projects = false
+        project_id = dbt_cloud_project.test_project.id
+    }
+}
+
+// permission_set accepts one of the following values:
+// "account_admin","admin","database_admin","git_admin","team_admin","job_admin","job_viewer","analyst","developer","stakeholder","readonly","project_creator","account_viewer","metadata_only"
```
Comment (GtheSheep) [inline examples/resources/dbt_cloud_service_token/resource.tf:15]: nit: New line at EOF

File: pkg/dbt_cloud/service_token.go [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	body, err := c.doRequest(req)
+	if err != nil {
+		return nil, err
+	}
+
+	groupResponse := GroupResponse{}
```
Comment (GtheSheep) [inline pkg/dbt_cloud/service_token.go:146]: is this right?

Comment (b-per) [reply pkg/dbt_cloud/service_token.go]: No, it isn't. I updated the code.

**Comment by b-per** [comment]
Fixing the merge conflict in 2 min

**Comment by b-per** [comment]
The conflict is fixed.

**Review by GtheSheep — APPROVED** [review]

**Pull request merged**